mood-analysis.rb

```diff
@@ -19,6 +19,52 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def strip_punctuation(sentence)
+  sentence.gsub(/[[:punct:]]/, "")
+end
+
+def happy_days(journal)
+  entries = 0
+  happy_count = 0
+  # wanted to do this with reduce but didn't get it figuerd out :/
+  journal.each do |entry|
+    happy_count += 1 if analyze_mood(strip_punctuation(entry)) == ":-)"
+    entries += 1
+    if happy_count > 3
+      break
+    end
+  end
+  if happy_count < 3
+    puts "In all #{entries} enterest, there were not 3 happy days! Sorry :-(."
+  else
+    puts "It takes #{entries} entries for 3 happy days to occur."
+  end
+end
+
+def overall_mood(journal)
+  entry_moods_frequency = {":-)" => 0, ":-(" => 0, ":-|" => 0}
+
+  journal.each do |entry|
+    mood = analyze_mood(strip_punctuation(entry))
+    case mood
+    when ":-)"
+      entry_moods_frequency[":-)"] += 1
+    when ":-("
+      entry_moods_frequency[":-("] +=1
+    when ":-|"
+      entry_moods_frequency[":-|"] +=1
+    end
+
+  end
```

Comment:
This works. But isn't easily scalable. there's no reason for us to be specific about what emoticons we are looking for or breaking out the individual cases. instead you can say:

```
  emoticons = Hash.new(0)
  entries.each do |entry|
    emoticon = analyze_mood(entry)
    emoticons[emoticon] += 1
  end
```

Well that's what I did anyway. To make it more general and then later if we add more emoticons they are automatically counted without any rewriting of code here.